src/foam/blob/HttpServletRequestBlob.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.blob;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.BufferedInputStream;
+import java.nio.ByteBuffer;
+
+public class HttpServletRequestBlob
+    extends foam.blob.AbstractBlob
+{
+  protected long size_;
+  protected long pos_ = 0;
+  protected HttpServletRequest request_;
+
+
+  public HttpServletRequestBlob(HttpServletRequest request) {
+    this.request_ = request;
+    this.size_ = request.getContentLengthLong();
+  }
+
+  @Override
+  public Buffer read(Buffer buffer, long offset) {
+    try {
+      if ( offset != pos_ ) {
+        throw new Exception("Offset does not match stream position");
+      }
+
+      int outOffset = 0;
+      long length = Math.min(buffer.getLength(), getSize() - offset);
+      if (length < buffer.getLength()) {
+        buffer = buffer.slice(0, length);
+      }
+
+      ByteBuffer bb = buffer.getData();
+      BufferedInputStream bufferedInputStream = new BufferedInputStream(request_.getInputStream());
```

Comment:
I don't think we should get the input stream every time, let's store it in a private/protected member variable and read from it.

We shouldn't do this decoration with BufferedInputStream every time .read() is called.  We should only do it once and store it in a member variable.

In fact we should fix this in FileReader as well.

Reply:
Thanks. I'll change that. So then there's no point in storing the whole request as a member variable then. Just the input stream and the content length.